### PR TITLE
Fix Ubuntu-System-CI

### PR DESCRIPTION
**Pull request by ryan-bournes**
Change number of diffusion iterations, boundary sizes and resolution of EulerDirichletBoundaries. 
Changed number of diffusion iterations for EulerNeumannZeroBoundaries. 
Changed number of diffusion iterations for EulerNeumannNonZeroBoundaries.

### CODE REVIEW

**Comment by TobiasDuswald** [comment]
Hey @ryan-bournes , 

I saw you assigned me and @nicogno as reviewers. The PR is supposed to fix the Ubuntu System CI, however, it still fails. Please take a look at the log. You'll probably have to add to `biodynamo/util/valgrind-bdm.supp` to fix it. See previous changes to this file for reference.

**Comment by ryan-bournes** [comment]
Hi @TobiasDuswald , the changes seem to have passed the ubuntu checks but has failed the macOS and Centos checks, will this be an issue?

File: test/unit/core/diffusion_test.cc [hunk 1]
```diff
@@ -974,8 +974,8 @@ TEST(DiffusionTest, EulerDirichletBoundaries) {
   double simulation_time_step{0.1};
   auto set_param = [](auto* param) {
     param->bound_space = Param::BoundSpaceMode::kClosed;
-    param->min_bound = -100;
-    param->max_bound = 100;
+    param->min_bound = -50;
+    param->max_bound = 50;
```
Comment (TobiasDuswald) [inline test/unit/core/diffusion_test.cc:978]: This modification looks good to me. The test verifies that with dirichlet = 1, at some point all voxels converge to 1. Making the domain smaller speeds up the test.

File: test/unit/core/diffusion_test.cc [hunk 2]
```diff
@@ -1045,7 +1045,7 @@ TEST(DiffusionTest, EulerNeumannZeroBoundaries) {
     rm->AddContinuum(dgrid);
 
     // Simulate diffusion / exponential decay for `tot` timesteps
-    int tot = 10000;
+    int tot = 1000;
```
Comment (TobiasDuswald) [inline test/unit/core/diffusion_test.cc:1048]: Have you checked if the concentration still reaches the boundary @ryan-bournes ? This test is supposed to verify that nothing escapes with homogeneous Neumann conditions, the test is only meaningful if the concentration reaches the boundaries. Possibly a good idea to lower `int tot` but add a source close to each of the six boundaries.

File: test/unit/core/diffusion_test.cc [hunk 3]
```diff
@@ -1104,7 +1104,7 @@ TEST(DiffusionTest, EulerNeumannNonZeroBoundaries) {
   // ----------------------------------------------------------
 
   double intermediate_concentration_backup = 0.0;
-  int tot = 10000;
+  int tot = 1000;
```
Comment (TobiasDuswald) [inline test/unit/core/diffusion_test.cc:1107]: I think this modification is good.

File: util/valgrind-bdm.supp [hunk 4]
```diff
@@ -529,3 +529,44 @@
    fun:__cxa_finalize
    ...
 }
+
```
Comment (TobiasDuswald) [inline util/valgrind-bdm.supp:532]: ✅

**Review by TobiasDuswald — COMMENTED** [review]
Thanks @ryan-bournes . Can you double check that the one unit test still does what it is supposed to do (see review comment). Issues on CentOS / macOS can be ignored for now (except if they result from the unit test changes of course).

**Comment by TobiasDuswald** [comment]
@nicogno if you found time to check this PR that would be great, it affects the unit tests you designed in the scope of the boundary conditions :)

**Comment by ryan-bournes** [comment]
@TobiasDuswald I expect this test to pass regardless but I have modified it so the sources are right next to the boundaries.

**Comment by nicogno** [comment]
Hi @ryan-bournes, I agree with @TobiasDuswald on the comments about the changes! Everything looks good to me, as long as you double check that the concentration reaches the boundaries in the Neumann Zero Boundaries test

**Comment by nicogno** [comment]
> Hi @TobiasDuswald , the changes seem to have passed the ubuntu checks but has failed the macOS and Centos checks, will this be an issue?

It looks like they're affected by some missing dependency.

File: test/unit/core/diffusion_test.cc (hunk 1 above)
Comment (nicogno) [reply test/unit/core/diffusion_test.cc:978]: Looks good to me, the logic behind the test wasn't altered.

File: test/unit/core/diffusion_test.cc (hunk 3 above)
Comment (nicogno) [reply test/unit/core/diffusion_test.cc:1107]: Looks good to me as well, total diffusion time should be enough for the test.

File: util/valgrind-bdm.supp (hunk 4 above)
Comment (nicogno) [reply util/valgrind-bdm.supp:532]: Can you elaborate more on this? Is this temporary?

**Comment by TobiasDuswald** [comment]
> > Hi @TobiasDuswald , the changes seem to have passed the ubuntu checks but has failed the macOS and Centos checks, will this be an issue?
> 
> It looks like they're affected by some missing dependency.

yes, none of the concerns for this PR. See also development branch related to #356 .

**Review by TobiasDuswald — APPROVED** [review]
Looks good to me, thanks @ryan-bournes ! 🚀

File: util/valgrind-bdm.supp (hunk 4 above)
Comment (TobiasDuswald) [reply util/valgrind-bdm.supp:532]: The file captures memory leaks and issues detected by valgrind in underlying libraries that we don't have any influence on. This should be fine. 👍

Comment (nicogno) [reply util/valgrind-bdm.supp:532]: All right, got it! Thanks

**Review by nicogno — APPROVED** [review]
Looks good to me! Thank you @ryan-bournes
